dbflow/src/main/java/com/raizlabs/android/dbflow/sql/saveable/ModelSaver.java

```diff
@@ -113,4 +119,13 @@ boolean delete(ModelAdapter<TModel> modelAdapter, TAdapter adapter, TTable model
         adapter.updateAutoIncrement(model, 0);
         return successful;
     }
+
+    private void closeInsertStatementAsync(final DatabaseStatement insertStatement) {
+        CLEAN_UP_HANDLER.post(new Runnable() {
```

Comment:
Lets store this as a variable property of the class, and each call to this method would check and re-`post()` the `Runnable` for future closing. We should also `cancel()` the runnable if it is scheduled for closing before using the `InsertStatement`. Since we have only one `ModelSaver` per table and they're not shared there will be minor conflict.